src/refinement-list/refinement-list.ts

```diff
@@ -70,17 +77,18 @@ export class NgAisRefinementList extends BaseWidget {
   // render options
   @Input() public showMoreLabel: string = 'Show more';
   @Input() public showLessLabel: string = 'Show less';
-  @Input() public transformItems?: Function;
   @Input() public searchable?: boolean;
   @Input() public searchPlaceholder: string = 'Search here...';
 
   // connectors options
   @Input() public attribute: string;
   @Input() public operator: 'or' | 'and' = 'or';
-  @Input() public limit: number | string = 10;
-  @Input() public showMoreLimit: number | string;
-  @Input() public showMore: boolean = false;
+  @Input() public limit: number = 10;
+  @Input() public showMore?: boolean;
+  @Input() public showMoreLimit: number;
   @Input() public sortBy: string[] | ((item: object) => number);
+  @Input()
+  public transformItems?: (items: RefinementListItem[]) => RefinementListItem[];
```

Comment:
@samouss and all typescript lovers 
i was tempted to write `(items: RefinementListItem[]) => any[];` since users can add properties to the object.
I think we should at least make sure the required keys are there.
Does this mean we should make `RefinementListItem` an interface ? Could something like `(items: RefinementListItem[]) => (any & RefinementListItem)[]` work ?

Reply:
any & is the same as just any

Reply:
I think `RefinementListItem & { [key: any]: any }` might be the best we can find 🤔

Reply:
You can use something similar to the `map` signature: 

```ts
<U extends RefinementListItem>(items: RefinementListItem[]) => U[]`
```

Note that the `extends` is optional, it ensures that the output won't break our template. But if at some point we expose something similar to slots users might want to return something that don't match `RefinementListItem`.

Reply:
perfect, thanks!